smartsim/settings/launchSettings.py

```diff
@@ -0,0 +1,407 @@
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import annotations
+import typing as t
+import copy
+
+
+from smartsim.log import get_logger
+from .._core.utils.helpers import fmt_dict
+from .common import validate_env_vars, validate_args, StringArgument
+from .launchCommand import LauncherType
+from .translators.launch.alps import AprunArgTranslator
+from .translators.launch.lsf import JsrunArgTranslator
+from .translators.launch.mpi import MpiArgTranslator, MpiexecArgTranslator, OrteArgTranslator
+from .translators.launch.pals import PalsMpiexecArgTranslator
+from .translators.launch.slurm import SlurmArgTranslator      
+from .translators.launch.dragon import DragonArgTranslator  
+from .translators.launch.local import LocalArgTranslator 
+from .translators import LaunchArgTranslator   
+from .baseSettings import BaseSettings                                                        
+
+logger = get_logger(__name__)
+
+class LaunchSettings(BaseSettings):
+    def __init__(
+        self,
+        launcher: t.Union[LauncherType, str],
+        launcher_args: t.Optional[t.Dict[str, t.Union[str,int,float,None]]] = None,
+        env_vars: t.Optional[StringArgument] = None,
+    ) -> None:
+        try:
+            self._launcher = LauncherType(launcher)
+        except KeyError:
+            raise ValueError(f"Invalid launcher type: {launcher}")
+        self._arg_translator = self._get_launcher()
+        
+        if env_vars:
+            validate_env_vars(env_vars)
+        self.env_vars = env_vars or {}
+
+        if launcher_args:
+            validate_args(launcher_args)
+        self.launcher_args = launcher_args or {}
```

Comment:
Per offline discussion: it was decided to remove this attr and have the translators manage their own internal representation of partially built settings args